### ISSUE TITLE
suggestions

### ISSUE THREAD

**Issue by mli** [zackchase/mxnet-the-straight-dope#1]
1. ndarray

   - slicing, one can write multiple dimensions, such as `x[2:3, 3:4] = 2`
   - context. cpu(0) and cpu(1) are the same, both of them represent all cpu cores. people may think cpu(0) means core 0. while gpu(0) means the 0-th gpu. as discussed before, we should add a `try_gpu_ctx()` 

2. autograd
   - beggining of head gradients. you are using x, y and z in text, while x,y,f in codes
   - end: should `print(a.grad)`

3. lr scrach
   - loss function, can we use `(yhat-y)**2`? 
   - last cell
     - can we move unncessary staffs out of `record()`. 
     - if we set `moving_loss=0` within the first for loop, we don't needs to check `if i==0`. 
     -  we don't need to call `asnumpy()`, i remember there is nd.mean(), and then call `asscalar()`. it should be better on performance if running on gpu.

another question is, will you also add a softmax-gluon?

**Comment by zackchase** [comment]
Addressed almost all concerns. Will revisit from scratch before closing ticket.

**Comment by zackchase** [comment]
Fixed all issues (I think...).
